scenarios/cpp/windows/captioning/captioning.cpp

```diff
@@ -0,0 +1,538 @@
+                    WriteToConsoleOrFile(CaptionFromSpeechRecognitionResult(sequenceNumber, e.Result));
+                }
+                else if (ResultReason::NoMatch == e.Result->Reason)
+                {
+                    WriteToConsole("NOMATCH: Speech could not be recognized.\n");
+                }
+            });
+
+        speechRecognizer->Canceled.Connect([this, &recognitionEnd](const SpeechRecognitionCanceledEventArgs& e)
+            {
+                if (CancellationReason::EndOfStream == e.Reason)
+                {
+                    WriteToConsole("End of stream reached.\n");
+                    recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+                }
+                else if (CancellationReason::CancelledByUser == e.Reason)
+                {
+                    WriteToConsole("User canceled request.\n");
+                    recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+                }
+                else if (CancellationReason::Error == e.Reason)
+                {
+                    std::ostringstream error;
+                    error << "Encountered error.\n"
+                        << "ErrorCode: " << (int)e.ErrorCode << "\n"
+                        << "ErrorDetails: " << e.ErrorDetails << std::endl;
+                    recognitionEnd.set_value(std::optional<std::string>{ error.str() }); // Notify to stop recognition.
+                }
+                else
+                {
+                    std::ostringstream error;
+                    error << "Request was cancelled for an unrecognized reason: " << (int)e.Reason << std::endl;
+                    recognitionEnd.set_value(std::optional<std::string>{ error.str() }); // Notify to stop recognition.
+                }
+            });
+
+        speechRecognizer->SessionStopped.Connect([this, &recognitionEnd](const SessionEventArgs& e)
+            {
+                WriteToConsole("Session stopped.\n");
+                recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+            });
+
+        // Starts continuous recognition. Uses StopContinuousRecognitionAsync() to stop recognition.
+        speechRecognizer->StartContinuousRecognitionAsync().get();
+
+        // Waits for recognition end.
+        auto result = recognitionEnd.get_future().get();
+
+        // Stops recognition.
+        speechRecognizer->StopContinuousRecognitionAsync().get();
+        
+        return result;
+    }
+};
+
+int main(int argc, char* argv[])
+{
+    const std::string usage = "Usage: caption.exe [...]\n\n"
+"  CONNECTION\n"
+"    --key KEY                     Your Azure Speech service subscription key.\n"
+"    --region REGION               Your Azure Speech service region.\n"
+"                                  Examples: westus, eastus\n\n"
+"  LANGUAGE\n"
+"    --languages LANG1,LANG2       Enable language identification for specified languages.\n"
```

Comment:
Could this go under the recognition heading? Just double checking is it comma separated or semicolon like phrase list?

Reply:
Good point. I thought the service expected a comma delimited list but no, it wants an array, so we can use any delimiter. I'll use ; instead to match SPX.